### PR TITLE
designs availability schema

### PR DESCRIPTION
**Pull request by GurkirtSingh**
### What this PR does (required):
- Creates **Schedule** model to store the weak based availability
- Adds **active schedule** field to **Profile** model

### Any information needed to test this feature (required):
- N/A

### Any issues with the current functionality (optional):
- N/A

### CODE REVIEW

**Review by sundayezeilo — COMMENTED** [review]
Schema looks good! But please always link your PR alongside filling the PR templates.

File: server/models/Schedule.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  profileId: {
+    type: mongoose.Schema.Types.ObjectId,
+    required: true,
+    ref: "Profile",
+  },
+  name: {
+    type: String,
+    default: "",
+  },
+  days: [
```
Comment (rajivtitus) [inline server/models/Schedule.js]: Let's say `min` and `max` indicate the days of the week, this design will allow the user to create redundant day schedules. Please see below for a suggestion:

```
const daySchedule = new mongoose.Schema({
  day: {
        type: Number,
        min: 1,
        max: 7,
        unique: true,
        required: true,
      },
  isAvailable: {
        type: Boolean,
        default: false,
      },
  start: {
        type: Number,
        min: 0,
        max: 23,
        required: true,
      },
  end: {
        type: Number,
        min: 0,
        max: 23,
        required: true,
      },
})

const scheduleSchema = new mongoose.Schema({
  profileId: {
    type: mongoose.Schema.Types.ObjectId,
    required: true,
    ref: "Profile",
  },
  name: {
    type: String,
    default: "",
  },
  days: {
    Sunday: daySchedule,
    Monday: daySchedule,
    ........
}
})


```

Comment (GurkirtSingh) [reply server/models/Schedule.js]: I just come across this problem and was thinking to limit the array size to 7. But your suggestion is great 😊.

**Review by sundayezeilo — COMMENTED** [review]
Thank you for making the requested changes :+1:

File: server/models/Schedule.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+const daySchedule = new mongoose.Schema({
+  isAvailable: {
+    type: Boolean,
+    default: false,
+  },
+  start: {
+    type: Number,
+    min: 0,
+    max: 23,
+    required: function () {
```
Comment (rajivtitus) [inline server/models/Schedule.js:12]: 👍🏽

**Review by rajivtitus — APPROVED** [review]
Excellent work! This schema design will fit our project use case now 💯

**Pull request merged**